bundles/EcommerceFrameworkBundle/IndexService/Worker/ElasticSearch/AbstractElasticSearch.php

```diff
@@ -836,4 +840,126 @@ protected function createEsIndex(string $indexName)
         $this->settingsHash = md5(json_encode($this->tenantConfig->getIndexSettings()));
         $this->updateVersionFile();
     }
+
+    /**
+     * Get the next index version, e.g. if currently 13, then 14 will be returned.
+     * @return int
+     */
+    private function getNextIndexVersion() : int {
+        return $this->getIndexVersion()+1;
+    }
+
+    /**
+     * Build the name of the index for a specific index number.
+     * @param int $id e.g. 13
+     * @return string the name of the index, such as at_de_elastic_13
+     */
+    private function buildIndexName(int $id) : string {
+        $currentIndexVersion = $this->getIndexVersion();
+        $this->indexVersion = $id;
+        $indexName = $this->getIndexNameVersion();
+        $this->indexVersion = $currentIndexVersion;
+        return $indexName;
+    }
+
+    /**
+     * Create an index with the given name (delete existing ones), and force a rebuild in ES.
+     * Does not perform an index switch.
+     * @param string $indexName
+     * @throws BadRequest400Exception
+     * @throws \Elasticsearch\Common\Exceptions\NoNodesAvailableException
+     */
+    private function buildIndexAndPerformEsRebuild(string $indexName) {
+        $indexName = strtolower($indexName);
+        $esClient = $this->getElasticSearchClient();
+        Logger::info('Index '.$indexName.' already existed. Delete and rebuild it...');
+        $result = $esClient->indices()->exists(['index' => $indexName]);
+        if ($result) {
+            Logger::info('Deleted '.$indexName.'.');
+            $esClient->indices()->delete(['index' => $indexName]);
+        }
+
+        $this->createEsIndex($indexName);
```

Comment:
this will result in inconsistent version file!! 
new index is created, but index number in version file is updated (since `$this->indexVersion` is not changed), but settings has is updated in version file.

Reply:
in `performNativeReindexing` the version file is then updated correctly, but `buildIndexAndPerformEsRebuild` must not be called without that context ... so maybe we should move that code to `performNativeReindexing`?